hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -2193,6 +2193,13 @@ void removeKeys(
    */
   @Retries.RetryTranslated
   public boolean delete(Path f, boolean recursive) throws IOException {
+    if (f.isRoot()) {
```

Comment:
this should go after the entrypoint, so that the metrics are incremented and FS open/closed state are checked